### PR TITLE
In the object stores, use the serialization strategy defined on the runtime type 

### PR DESCRIPTION
**Pull request by mgasner**

### CODE REVIEW

File: python_modules/dagster/dagster/core/object_store.py [hunk 1]
```diff
@@ -172,7 +174,10 @@ def get_object(self, context, runtime_type, paths):
 
         key = self._key_for_paths(paths)
 
-        return pickle.loads(self.s3.get_object(Bucket=self.bucket, Key=key)['Body'].read())
+        return runtime_type.serialization_strategy.deserialize_value(
+            # pylint: disable=protected-access
```
Comment (schrockn) [inline python_modules/dagster/dagster/core/object_store.py]: what's up with this _raw_stream thing?

File: python_modules/dagster/dagster_tests/core_tests/test_object_store.py [hunk 2]
```diff
@@ -52,3 +94,35 @@ def test_s3_object_store():
...
+            .read()
+            .decode('utf-8')
+            == 'FOO'
+        )
+
+        assert object_store.has_object(context, ['foo'])
+        assert object_store.get_object(context, LowercaseString.inst(), ['foo']) == 'foo'
+    finally:
+        try:
+            shutil.rmtree(object_store.root)
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/core_tests/test_object_store.py]: very confused by this

File: python_modules/dagster/dagster_tests/core_tests/test_object_store.py [hunk 3]
```diff
@@ -29,7 +47,30 @@ def test_file_system_object_store():
...
+
+        with open(os.path.join(object_store.root, 'foo'), 'rb') as fd:
+            assert fd.read().decode('utf-8') == 'FOO'
+
+        assert object_store.has_object(context, ['foo'])
+        assert object_store.get_object(context, LowercaseString.inst(), ['foo']) == 'foo'
+    finally:
+        try:
+            shutil.rmtree(object_store.root)
+        except (seven.FileNotFoundError, OSError):
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/core_tests/test_object_store.py]: maybe wrap all this up in a context manager? along with comment about doing nothing in the case is ok. Also what error does OSError represent here?

File: python_modules/dagster/dagster_tests/core_tests/test_object_store.py (hunk 2 above)
Comment (mgasner) [reply python_modules/dagster/dagster_tests/core_tests/test_object_store.py]: this is just test case cleanup

File: python_modules/dagster/dagster_tests/core_tests/test_object_store.py [hunk 4]
```diff
@@ -4,12 +4,30 @@
...
 from dagster.core.execution import yield_pipeline_execution_context
+from dagster.core.types.marshal import SerializationStrategy
 from dagster.core.object_store import FileSystemObjectStore, S3ObjectStore
-from dagster.core.types.runtime import Bool
+from dagster.core.types.runtime import Bool, RuntimeType
 
 from ..marks import aws, nettest
 
 
+class UppercaseSerializationStrategy(SerializationStrategy):
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/core_tests/test_object_store.py]: haha nice

File: python_modules/dagster/dagster_tests/core_tests/test_object_store.py (hunk 2 above)
Comment (schrockn) [reply python_modules/dagster/dagster_tests/core_tests/test_object_store.py]: i think you fixed this by making it rm_object actually

**Pull request merged**